rules/integrations/aws/credential_access_rapid_secret_retrieval_attempts_from_secretsmanager.toml

```diff
@@ -0,0 +1,74 @@
+[metadata]
+creation_date = "2024/04/11"
+integration = ["aws"]
+maturity = "production"
+min_stack_comments = "AWS integration breaking changes, bumping version to ^2.0.0"
+min_stack_version = "8.9.0"
+updated_date = "2024/04/11"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule attempts to identify rapid secret retrieval attempts from AWS SecretsManager. Adversaries may attempt to
+retrieve secrets from the Secrets Manager programmatically using the `GetSecretValue` or `BatchGetSecretValue` API
+actions.
+"""
+false_positives = [
+    """
+    Verify whether the user identity, user agent, and/or hostname should be using GetSecretString or BatchGetSecretValue
+    APIs for the specified SecretId. If known behavior is causing false positives, it can be exempted from the rule.
+    """,
+]
+from = "now-9m"
+index = ["filebeat-*", "logs-aws.cloudtrail*"]
+language = "kuery"
+license = "Elastic License v2"
+name = "Rapid Secret Retrieval Attempts from AWS SecretsManager"
+references = [
+    "https://docs.aws.amazon.com/secretsmanager/latest/apireference/API_GetSecretValue.html",
+    "https://detectioninthe.cloud/ttps/credential_access/access_secret_in_secrets_manager/",
+    "https://cloud.hacktricks.xyz/pentesting-cloud/aws-security/aws-services/aws-secrets-manager-enum",
+]
+risk_score = 47
+rule_id = "185c782e-f86a-11ee-9d9f-f661ea17fbce"
+severity = "medium"
+tags = [
+    "Domain: Cloud",
+    "Data Source: AWS",
+    "Data Source: Amazon Web Services",
+    "Data Source: AWS Secrets Manager",
+    "Tactic: Credential Access",
+    "Resources: Investigation Guide",
+]
+timestamp_override = "event.ingested"
+type = "threshold"
+
+query = '''
+event.dataset:aws.cloudtrail and event.provider:secretsmanager.amazonaws.com and
+    event.action: (GetSecretValue or BatchGetSecretValue) and event.outcome:success and
+    not user_agent.name: ("Chrome" or "Firefox" or "Safari" or "Edge" or "Brave" or "Opera")
+'''
+
+
+[[rule.threat]]
+framework = "MITRE ATT&CK"
+[[rule.threat.technique]]
+id = "T1555"
+name = "Credentials from Password Stores"
+reference = "https://attack.mitre.org/techniques/T1555/"
+[[rule.threat.technique.subtechnique]]
+id = "T1555.006"
+name = "Cloud Secrets Management Stores"
+reference = "https://attack.mitre.org/techniques/T1555/006/"
+
+
+
+[rule.threat.tactic]
+id = "TA0006"
+name = "Credential Access"
+reference = "https://attack.mitre.org/tactics/TA0006/"
+
+[rule.threshold]
+field = ["aws.cloudtrail.user_identity.arn"]
+value = 10
```

Comment:
10 in 9m seems to be too slow, maybe adjust/reduce the `from = "now-9m"` window

Reply:
good point. I doubled to retrieval attempts to 20 and changed this to user.id to account for AWS services, not just IAM users.